### PR TITLE
Substitute defines DEG_TO_RAD / RAD_TO_DEG with ofDegToRad / ofRadToDeg

### PR DESCRIPTION
**Pull request by dimitre**
More
Substituting MIN / MAX  to std::min std::max
FLT_EPSILON to std equivalent
pi, half_pi, two_pi by glm ones

now we can remove the unused constants later on a breaking release 0.12.0

### CODE REVIEW

File: libs/openFrameworks/gl/ofLight.cpp [hunk 1]
```diff
@@ -224,7 +224,7 @@ float ofLight::getSpotlightCutOff() const{
 
 //----------------------------------------
 void ofLight::setSpotConcentration( float exponent ) {
-    data->exponent = CLAMP(exponent, 0, 128);
+    data->exponent = ofClamp(exponent, 0, 128);
```
Comment (artificiel) [inline libs/openFrameworks/gl/ofLight.cpp:227]: how about std::clamp? it is smarter with types. actually perhaps ofClamp should be deprecated?

File: libs/openFrameworks/math/ofMath.h [hunk 2]
```diff
@@ -284,7 +284,7 @@ float ofRadToDeg(float radians);
 ///
 /// \param degrees An angle in degrees.
 /// \returns the angle in radians.
-float ofDegToRad(float degrees);
+float ofDegToRad(const float & degrees);
```
Comment (artificiel) [inline libs/openFrameworks/math/ofMath.h:287]: will this &ref not remove the possibility of writing directly 

```c++
auto a = ofDegToRad(90);
```

Comment (dimitre) [reply libs/openFrameworks/math/ofMath.h]: totally agreed. reverted and pushed a commit

File: libs/openFrameworks/gl/ofLight.cpp (hunk 1 above)
Comment (artificiel) [reply libs/openFrameworks/gl/ofLight.cpp:227]: ah! yes it's c++17, but I thought it was allowed for 0.12?

**Comment by ofTheo** [comment]
looks great @dimitre!

File: libs/openFrameworks/3d/ofCamera.cpp [hunk 3]
```diff
@@ -60,7 +60,7 @@ void ofCamera::setupPerspective(bool _vFlip, float fov, float nearDist, float fa
 	ofRectangle orientedViewport = getRenderer()->getNativeViewport();
 	float eyeX = orientedViewport.width / 2;
 	float eyeY = orientedViewport.height / 2;
-	float halfFov = PI * fov / 360;
+	float halfFov = PI * fov / 360.0f;
```
Comment (ofTheo) [inline libs/openFrameworks/3d/ofCamera.cpp:63]: any reason to not replace PI internally with `glm::pi<float>()` ? 
( but still keep the define for legacy use )

Comment (dimitre) [reply libs/openFrameworks/3d/ofCamera.cpp:63]: Yeah we can do it. I'll substitute PI and TWO_PI / M_TWO_PI by the glm constants.
maybe we have to have some more glm includes

Comment (dimitre) [reply libs/openFrameworks/3d/ofCamera.cpp]: done in latest commit, passing tests. should we merge it?

**Comment by ofTheo** [comment]
Hey!  I think there are two fixes need at lines 511 and 512 of of3DGraphics. Pi is used in instead of two_pi 

https://github.com/openframeworks/openFrameworks/pull/7413/commits/6783807e264ca01e2f8daab4de040256507b1eeb#diff-d65e3c3d754334a7189b5c27b80d2f6c4760d9357b1c5b8639c99082ab1acfeeR511

The rest looks great!

File: libs/openFrameworks/graphics/ofGraphics.cpp [hunk 4]
```diff
@@ -398,11 +398,11 @@ void ofBackgroundGradient(const ofColor& start, const ofColor& end, ofGradientMo
 		gradientMesh.addVertex(glm::vec3(center, 0.f));
 		gradientMesh.addColor(start);
 		int n = 32; // circular gradient resolution
-		float angleBisector = TWO_PI / (n * 2);
+		float angleBisector = glm::two_pi<float>() / (n * 2);
```
Comment (NickHardeman) [inline libs/openFrameworks/graphics/ofGraphics.cpp:401]: Do all of the numbers need to be floats? Same for line 405 in ofGraphics.cpp
float angleBisector = glm::two_pi<float>() / ((float)n * 2.0f);

File: libs/openFrameworksCompiled/project/osx/config.osx.default.mk [hunk 5]
```diff
@@ -78,7 +78,7 @@ ifndef MAC_OS_STD_LIB
 endif
 
 ifndef MAC_OS_CPP_VER
-    MAC_OS_CPP_VER = -std=c++17
+    MAC_OS_CPP_VER = -std=c++20
```
Comment (ofTheo) [inline libs/openFrameworksCompiled/project/osx/config.osx.default.mk:81]: @dimitre was this meant to slip in? 🙂

Comment (dimitre) [reply libs/openFrameworksCompiled/project/osx/config.osx.default.mk:81]: oh no! 😓I'll be reverting now in a new PR

Comment (dimitre) [reply libs/openFrameworksCompiled/project/osx/config.osx.default.mk:81]: done and pushed to master. good catch @ofTheo thanks

**Pull request merged**